app/containers/ITunes/index.js

```diff
@@ -0,0 +1,136 @@
+/**
+ *
+ * ITunes
+ *
+ */
+
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { injectIntl, FormattedMessage as T } from 'react-intl';
+
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import { Input } from 'antd';
+import styled from 'styled-components';
+import get from 'lodash/get';
+import debounce from 'lodash/debounce';
+import isEmpty from 'lodash/isEmpty';
+
+import { selectITunes, selectSongData, selectSongError, selectSongName } from './selectors';
+
+import { iTunesCreators } from './reducer';
+import iTunesSaga from './saga';
+import { injectSaga } from 'redux-injectors';
+import ItunesCard from '@app/components/ItunesCard/index';
+
+const { Search } = Input;
+
+const GridLayout = styled.div`
+  && {
+    margin: 1.25em 0;
+    width: 100%;
+    display: flex;
+    flex-wrap: wrap;
+    justify-content: center;
+  }
+`;
+
+const Container = styled.div`
+  && {
+    width: 100%;
+    margin: 0 auto;
+    padding: 1.25em;
+  }
+`;
+
+export function ITunes({ songData = [], songError = null, songName, dispatchSongs, dispatchClearSongs }) {
+  const [loading, setLoading] = useState(false);
+  useEffect(() => {
+    const loaded = get(songData, 'results', null) || songError;
```

Comment:
```
 const loaded = get(songData, 'results', songError);
```


if `loaded` is dependent on both `songData` as well as `songError` the dependency array should have both of them.

```
  }, [songData, songError]);
```